.github/workflows/gradle.yml

```diff
@@ -13,44 +13,54 @@ jobs:
       matrix:
         java: ['11', '14']
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Set up JDK
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: 'adopt'
           java-version: ${{ matrix.java }}
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          # Publish a Gradle build scan so you can see all the build logs,
+          # a complete task timeline, test outputs, and the resolved dependencies of your build.
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+          build-scan-terms-of-use-agree: "yes"
       - name: Run Build
         id: build
-        uses: gradle/gradle-build-action@v2
+        run: ./gradlew build
```

Comment:
The `gradle/actions/setup-gradle` action replaces `gradle/gradle-build-action`.
Reference:
https://github.com/gradle/actions/tree/af6e5767241073b8ab20cc8cf7e95ec63fadc160/setup-gradle#the-setup-gradle-action